[Eun777] iP

Eun777 opened the pull request:
# EunAI

> "If you fail to plan, you plan to fail." - Benjamin Franklin ([source](https://www.goodreads.com/quotes/460142-if-you-fail-to-plan-you-are-planning-to-fail))

With EunAI, plan ahead, so you can complete and succeed. EunAI is an easy to use chatbot for storing all your tasks, both mundane and significant ones (even to remind you to boil rice, so you don't get cooked yourself). It's

- Text-based
- Easy to use
- Like your own ***personal assistant***
but **FREE** (for now...😅)

### How to get EunAI:

1. Download it from [here](https://github.com/Eun777/ip).
2. Run locally on your device.
3. Add your to dos, deadlines and events.
4. Run EunAI to get an overview of your tasks, anytime, anywhere.

### Features of EunAI:

- [x] Add to dos, deadlines and events
- [x] Manage saved tasks
- [x] Find tasks through keyword search
- [ ] Reminders (future update for ~~all~~ paid subscription only)



### Facing problems with EunAI?
Check the following code in EunAI.java and ensure that it matches your local directory:
```ruby
public static void main(String[] args) {
    String userHome = System.getProperty("user.home");
    String filePath = userHome + "/ip/data/eunAI.txt";
    new EunAI(filePath).run();
}
```
Else, check that your command follows the following format:
Get list of tasks: `list`
Add to do: `todo read book`
Add deadline: `deadline return book /by 2/12/2025 1800`
Add event: `event book fair /from 14/3/2025 1300 /to 17/3/2025 1500`
Mark task: `mark 1`
Unmark task: `unmark 2`
Find: `find keyword`
Exit: `bye`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Eun777 committed:
Chatbot name changed to EunAI

Eun777 committed:
Add echo to EunAI

Eun777 committed:
Add list functionality

Eun777 committed:
Add indentations to EunAI replies

Eun777 committed:
Add basic functionality for Task class

Eun777 committed:
Change taskList type from List<String> to List<Task>

Eun777 committed:
Add basic functionality for mark

Eun777 committed:
Add unmark functionality for Task list

Eun777 committed:
Add Date, Deadline, Event, ToDo classes

Eun777 committed:
Add ToDo, Event and Deadline functionality and support

Eun777 committed:
Handle errors without exceptions usage

Eun777 committed:
Add exceptions for EunAI

Eun777 committed:
Add exception handling in EunAI class

Eun777 committed:
Remove extra libraries

Eun777 committed:
Change from array to ArrayList<Task>

Eun777 committed:
Add basic testing to EunAI

Eun777 committed:
Add command parser and change to enums

Eun777 committed:
Add Storage class to handle local data storage

Eun777 committed:
Add functionality to Storage class

Eun777 committed:
Add support for storage and parse functions

Eun777 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add support for storage and parse functions
  Add functionality to Storage class
  Add Storage class to handle local data storage

Eun777 committed:
Change dates from String to LocalDateTime

Eun777 committed:
Add Command, Ui classes for more OOP

Eun777 committed:
Add Command, TaskList and Ui classes for MoreOOP

Eun777 committed:
Add more integration with Ui functions

Eun777 committed:
Merge branch 'branch-MoreOOP'

* branch-MoreOOP:
  Add Command, Ui classes for more OOP

# Conflicts:
#	src/main/java/Command.java

Eun777 committed:
Add classes to eunAI package

Eun777 committed:
Add Javadoc for EunAI, DataParser, Storage, TaskList classes

Eun777 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

* upstream/add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Eun777 committed:
Fix syntax error in TaskList

Eun777 committed:
Fix error in retrieving previously stored data

Eun777 committed:
Fix bug on saving tasks locally

Eun777 committed:
Add find function for search for a keyword in the task description

Eun777 committed:
Add JavaDocs for eunAI.task package

Eun777 committed:
Add JavaDocs for eunAI.ui package

Eun777 committed:
Remove unused exception classes

Eun777 committed:
Add JavaDocs for eunAI.exception package

Eun777 committed:
Add JavaDocs for eunAI.command package

Eun777 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDocs for eunAI.command package
  Add JavaDocs for eunAI.exception package
  Remove unused exception classes
  Add JavaDocs for eunAI.ui package
  Add JavaDocs for eunAI.task package

Eun777 committed:
Add checkStyle to detect coding style violations

Eun777 committed:
Fix coding style violations in files

Eun777 committed:
Add basic JUnit tests for CommandParser and TaskList

Eun777 committed:
Add edge case tests for CommandParser

Eun777 committed:
Fix CommandParser for edge case failure

Eun777 committed:
Add JUnit test for Deadline class

Eun777 committed:
Fix bug in DeadlineTest test case

Eun777 committed:
Fix input incompatibility error in DateParser

jiantaooo commented on src/main/java/eunAI/EunAI.java:
```diff
@@ -0,0 +1,61 @@
+    private final Ui ui;
+
+    /**
+     * Contructs an EunAI chatbot instance with the specified filePath for task storage.
```
```suggestion
     * Constructs an EunAI chatbot instance with the specified filePath for task storage.
```
there's a type on the word 'constructs'

jiantaooo commented on src/main/java/eunAI/EunAI.java:
Your code is well-structured and seems to adhere to the coding standard. nice nice

jiantaooo commented on src/main/java/eunAI/Storage.java:
Your code is very structured and adheres to the Java coding standard. The private method makes the logic clear and easy to understand. The error handling is well-implemented and effectively ensures robustness.

jiantaooo commented on src/main/java/eunAI/TaskList.java:
nothing much to say here. everything seems good.

jiantaooo commented on src/main/java/eunAI/command/Command.java:
The code follows proper naming conventions, layout rules, and commenting guidelines. Good job! Every method includes error handling, making the code robust and user-friendly.

jiantaooo commented on src/main/java/eunAI/task/Task.java:
```diff
@@ -0,0 +1,97 @@
+     *
+     * @return {@code true} if the task is done, {@code false} otherwise.
+     */
+    public boolean taskStatus() {
```
i think the naming of this getter method should be something like getTaskStatus() or isDone()

jiantaooo commented on src/main/java/eunAI/task/Task.java:
its good. just a suggestion for a more conventional name for a method.

jiantaooo reviewed (commented):
Well-structured overall and follows good Java standards. Just a few minor suggestions.

Wrongian commented on src/main/java/eunAI/EunAI.java:
```diff
@@ -0,0 +1,61 @@
+     * @param filePath The file path where tasks are saved and loaded.
+     */
+    public EunAI(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        taskList = new TaskList(storage.loadSavedTasks());
```
I think in the other code, you used "this" to assign class member instance variables. Perhaps you want to keep this consistent?

Wrongian commented on src/main/java/eunAI/Storage.java:
```diff
@@ -0,0 +1,98 @@
+    }
+
+    /**
+     * Loads previously saved tasks from the file specified by {@code filePath}.
+     * If the file does not exist, an empty task list is returned.
+     * @return An {@code ArrayList} of tasks loaded from the file.
```
I think you may want to to leave an empty line between the description and parameter section?

Wrongian reviewed (commented):
Overall, pretty high code quality.
The variable names are really understandable. Coding standards are generally well adhered to. Spacing between logical code blocks is really well used to make the code well understandable. Just a couple of comments that you may want to work to make your code even more exceptional.

Eun777 committed:
Add base files for JavaFX

Eun777 committed:
Add basic functioning GUI using JavaFX

Eun777 committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add basic functioning GUI using JavaFX
  Add base files for JavaFX

Eun777 committed:
Rename from EunAI to EunAi to follow coding standard

Eun777 committed:
Change name of package from eunAI to eunai to follow coding standard

Eun777 committed:
Fix style in Command class to follow standard convention

Eun777 committed:
Update JavaDocs for EunAI class

Eun777 committed:
Update JavaDocs for TaskList class

Eun777 committed:
Update response for find function

Eun777 committed:
Add filter by task type as additional functionality

Eun777 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Update JavaDocs for TaskList class
  Update JavaDocs for EunAI class

Eun777 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Fix style in Command class to follow standard convention
  Change name of package from eunAI to eunai to follow coding standard
  Rename from EunAI to EunAi to follow coding standard

# Conflicts:
#	src/main/java/eunAI/EunAI.java

Eun777 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add filter by task type as additional functionality
  Update response for find function

# Conflicts:
#	src/main/java/eunai/TaskList.java
#	src/main/java/eunai/command/Command.java

Eun777 committed:
Fix bug in eunAI and MainWindow

Eun777 committed:
Package EunAI as JAR file

Eun777 committed:
Fix bug preventing JAR file generation

The application failed to run because the JAR was missing a
Main-Class entry in the manifest. This commit updates the
build configuration to correctly set the entry point.

Ensured that Launcher is used to start the JavaFX application
to avoid classpath issues.

Eun777 committed:
EunAi and Command: add assertions to enforce valid states

Certain methods in `EunAi` and `Command` assume that their inputs and
dependencies are valid, but these assumptions are not explicitly checked.

If these assumptions are violated, the program may fail in unexpected ways,
making debugging harder.

Let's add assertions to verify critical assumptions in these classes.

- Ensure `filePath` in `EunAi` constructor is not null or empty.
- Validate user input in `processUserInput`.
- Add assertions in `Command` methods to check task list and input parsing.

These assertions will help catch potential bugs early during development
without affecting production performance.

Eun777 committed:
Remove assert false from EunAI class

Eun777 committed:
Fix code quality issues for EunAi and Command classes

EunAi and Command classes had code quality issues that made the
code harder to maintain and understand. These included SLAP
violations, redundant logic, and inconsistent exception handling.

To improve code maintainability and readability:
- Extracted parsing logic into helper methods
  (`extractDeadlineDetails`, `extractFindKeyword`).
- Refactored `handleMark`, `handleUnmark`, and `handleDelete`
  to use `getTaskFromInput` for consistency.
- Improved exception handling by replacing `catch (Exception e)`
  with specific exceptions.
- Replaced magic numbers with named constants
  (`MARK_PREFIX_LENGTH`, etc.).
- Updated `handleExit` to provide clearer user feedback
  on task saving.

These changes improve readability, maintainability, and robustness.

Eun777 committed:
Fix code quality issues in DateParser class

Eun777 committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:
  Remove assert false from EunAI class
  EunAi and Command: add assertions to enforce valid states

Eun777 committed:
Merge pull request #1 from Eun777/branch-A-Assertions

A-Assertions

Eun777 committed:
Merge pull request #2 from Eun777/branch-A-CodeQuality

A-CodeQuality

Eun777 committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:
  Fix code quality issues in DateParser class
  Fix code quality issues for EunAi and Command classes

Eun777 committed:
Merge commit '08f3d8a0e412b2587146453cbda19d8074aaf4ed'

* commit '08f3d8a0e412b2587146453cbda19d8074aaf4ed':

Eun777 committed:
Add extension C-MassOps

Eun777 committed:
Refactor handleMassOps to improve SLAP and readability

- Extracted task index parsing into `extractTaskIndices()`
- Moved command prefix length logic to `getCommandPrefixLength()`
- Separated task processing into `processMassOperation()`
- Added `finalizeMassOpsResult()` to handle final output formatting
- Reduced method length to <30 lines for better maintainability

This improves code clarity, reduces nesting, and ensures better
separation of concerns.

Eun777 committed:
Merge branch 'branch-BCD-extensions'

* branch-BCD-extensions:
  Refactor handleMassOps to improve SLAP and readability
  Add extension C-MassOps

Eun777 committed:
Change UX and colour scheme to fit EunAI's colour scheme

Eun777 committed:
Make EunAI's welcome message more playful and engaging

The current welcome message is plain and lacks personality:
"Hello! I'm EunAI. How can I assist you today?"

A chatbot with a more lively and casual tone creates a more
engaging user experience. A fun and friendly greeting makes
interactions feel less robotic and more conversational.

Let's update the welcome message to:
"Hellooo, I'm EunAI! ✨ What's up? Need some help or just here
to vibe lol"

This change makes EunAI feel more approachable and engaging,
helping users feel more comfortable interacting with it.

Eun777 committed:
Improve SLAP and error handling in Command class

Eun777 committed:
Refactor code for Storage and Ui class

Eun777 committed:
Update Ui.png, README.md for user guide

Eun777 committed:
Update README.md with user guide for mark, unmark, delete, find

Eun777 committed:
Update user docs to include supported date formats

Eun777 committed:
Fix minor spacing issue in README.md

Eun777 committed:
Update user guide for EunAI to include 'bye' and MassOps commands

Eun777 committed:
Refactor code for OOP and SLAP requirements

Eun777 committed:
Refactor Command class for OOP and SLAP requirements

Eun777 committed:
Improve invalid message handling

Eun777 committed:
Update user guide for MassOps

Eun777 committed:
Reorganise dependencies

Eun777 committed:
Add JavaDocs for Command class

Eun777 committed:
Update README.md with Ui.png
